generate.go

```diff
@@ -19,14 +19,28 @@ limitations under the License.
 // NOTE: See the below link for details on what is happening here.
 // https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module
 
+//go:generate go run github.com/crossplane-contrib/terraform-provider-dl --config=terraform-provider-gen.yaml --output=tf-plugin
```

Comment:
I'm wondering if we should download this as part of generation or if it should be part of a different command. I am seeing the following in `make generate`, which seems like it should be separate (despite the error):

```
2021/04/13 14:53:45 [DEBUG] checking for provider in "tf-plugin"
2021/04/13 14:53:45 [DEBUG] found provider "terraform-provider-vsphere_v1.24.3_x4"
2021-04-13T14:53:45.538-0500 [INFO]  plugin: configuring client automatic mTLS
2021-04-13T14:53:45.581-0500 [DEBUG] plugin: starting plugin: path=/home/dan/code/github.com/crossplane-contrib/provider-terraform-vsphere/tf-plugin/terraform-provider-vsphere_v1.24.3_x4 args=[/home/dan/code/github.com/crossplane-contrib/provider-terraform-vsphere/tf-plugin/terraform-provider-vsphere_v1.24.3_x4]
terraform-provider-gen: error: Error while executing hiveworld command: Failed to initialize GRPC plugin: fork/exec /home/dan/code/github.com/crossplane-contrib/provider-terraform-vsphere/tf-plugin/terraform-provider-vsphere_v1.24.3_x4: permission denied
```